pkg/device/volume.go

```diff
@@ -147,3 +153,49 @@ func RemoveVolFinalizer(vol *apis.DeviceVolume) error {
 	_, err := volbuilder.NewKubeclient().WithNamespace(DeviceNamespace).Update(vol)
 	return err
 }
+
+// WaitForDeviceVolumeProcessed waits till the lvm volume becomes
+// ready or failed (i.e reaches to terminal state).
+func WaitForDeviceVolumeProcessed(ctx context.Context, volumeID string) (*apis.DeviceVolume, error) {
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, status.FromContextError(ctx.Err())
```

Comment:
NOTE: this code was copied over from [lvm-localpv](https://github.com/openebs/lvm-localpv/blob/master/pkg/lvm/volume.go#L115)

@iyashu This method needs to be removed as it was added [accidentally](https://github.com/grpc/grpc-go/pull/3561) in grpc as per [this release notes](https://github.com/grpc/grpc-go/releases/tag/v1.29.1).  

Didnt get the reason why, the method was used here, If I understand it better, I could replace it with the exact error that we intend to return to the caller.

Reply:
The build will pass only once this error is fixed.

Reply:
Done.